lib/grandpa/message_handler.go

```diff
@@ -284,27 +284,32 @@ func (h *MessageHandler) verifyCatchUpResponseCompletability(prevote, precommit
 	return nil
 }
 
-func getEquivocatoryVoters(votes []AuthData) map[ed25519.PublicKeyBytes]struct{} {
+func getEquivocatoryVoters(votes []AuthData) (map[ed25519.PublicKeyBytes]struct{}, error) {
 	eqvVoters := make(map[ed25519.PublicKeyBytes]struct{})
 	voters := make(map[ed25519.PublicKeyBytes]int, len(votes))
 
 	for _, v := range votes {
 		voters[v.AuthorityID]++
+		if voters[v.AuthorityID] > 2 {
+			return nil, errInvalidMultiplicity
+		}
 
-		if voters[v.AuthorityID] > 1 {
+		if voters[v.AuthorityID] == 2 {
```

Comment:
nit a switch could be nice:

```go
switch voters[v.AuthorityID] {
	case 1:
	case 2:
		eqvVoters[v.AuthorityID] = struct{}{}
	default:
		return nil, fmt.Errorf("%w: authority id %x has %d votes", 
			errInvalidMultiplicity, v.AuthorityID, voters[v.AuthorityID])
}
```

Reply:
I prefer a switch here as well.  Or at the very least an `else if` condition.